ground/src/main/java/com/google/android/ground/ui/startup/StartupFragment.kt

```diff
@@ -47,15 +45,18 @@ class StartupFragment : AbstractFragment() {
     savedInstanceState: Bundle?,
   ): View? = inflater.inflate(R.layout.startup_frag, container, false)
 
+  @Suppress("SwallowedException")
   override fun onResume() {
     super.onResume()
     showProgressDialog(R.string.initializing)
     viewLifecycleOwner.lifecycleScope.launch {
-      try {
-        viewModel.initializeLogin()
-      } catch (t: Throwable) {
-        onInitFailed(t)
-      }
+      val success =
+        try {
+          viewModel.initializeLogin()
+        } catch (t: Throwable) {
+          false
```

Comment:
viewModel.initializeLogin() doesn't log its own exceptions, so we should do that here or move the try/catch inside. In general, we should aim to use either `Boolean` or exceptions as a means of communicating invalid states, but not both. Ideally we'd just use exceptions and log them at the top level. Lmkwyt.